2_process/src/pair_nhd_catchments.R

```diff
@@ -68,9 +76,26 @@ get_intersecting_nhdplus_catchments <- function(prms_line,prms_hrus,min_area_ove
       # retain NHD catchments that overlap >= X% of their area with the HRU polygon
       filter(area_prop >= min_area_overlap)
     
+    # Download flowlines and VAA for COMIDs in nhd_cats_intrsct
+    nhd_flines_intrsct <- nhdplusTools::get_nhdplus(comid = c(nhd_cats_intrsct$featureid),
+                                                    realization = 'flowline') %>%
+      # reformat variable names to uppercase
+      rename_with(.,toupper,id:enabled)
+    
+    # If omit_zero_area_flines is TRUE, edit df to only return COMIDs where 
+    # attribute AREASQKM != 0 (unless there is only one COMID that drains to the 
+    # PRMS line, in which case keep the AREASQKM = 0 COMID to make sure that
+    # all PRMS/NHM segments are represented in the output data frame).
+    if(omit_zero_area_flines & length(nhd_flines_intrsct$COMID) > 1){
+      nhd_flines_out <- nhd_flines_intrsct %>%
+        filter(AREASQKM > 0) 
```

Comment:
I'm suggesting that the arbitrary value can be hard coded and informed by the maximum length for reaches with AREASQKM == 0 in our dataset. Hopefully this is something like 50 m because we have a 30 m DEM.

Reply:
The maximum length for reaches with `AREASQKM == 0` is 193 m and most likely represents a connector rather than a DEM artifact. I'm sure I'm missing something, but if we check reach lengths based on this maximum value, we shouldn't expect to return any flagged reaches (and so I wonder if this check is valuable, then)?

Reply:
> we shouldn't expect to return any flagged reaches

Yes, for this dataset. I thought the value of this check would be if we apply to other watersheds. I know this is a repo focused on DRB, so maybe not a good idea. 

Also, I didn't realize that we have longer reaches than would be expected by a DEM artifact. Maybe we set a 50 m arbitrary cutoff (assuming a 30 m DEM was used)?